x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -196,12 +196,12 @@ public static StepKey getCurrentStepKey(Settings indexSettings) {
         }
     }
 
-    static Step getCurrentStep(PolicyStepsRegistry stepRegistry, String policy, Index index, Settings indexSettings) {
+    static Step getCurrentStep(PolicyStepsRegistry stepRegistry, String policy, IndexMetaData indexMetaData, Settings indexSettings) {
         StepKey currentStepKey = getCurrentStepKey(indexSettings);
         if (currentStepKey == null) {
             return stepRegistry.getFirstStep(policy);
```

Comment:
As far as I can tell, this only needed to onboard an index. Can we move this (as a follow up) to a dedicated place? on boarding is an exceptional case - we should put it in it's own method.

Reply:
Yeah as a followup definitely, we'd need to remove it if we want to remove `PolicyStepsRegistry` too